crossplane/crossplane
# StackInstall source should be used in unqualified Stack deployments

### Description of your changes



When a StackInstall (or ClusterStackInstall) includes a `spec.source`, this value will be prepended to the Stack deployment so long as no other source string is detected.
    
StackDefinition resources also get this change, affecting where they locate behavior images and the generic templating-controller.
    
Fixes crossplaneio/crossplane#1246

_Approach is based on https://github.com/crossplaneio/crossplane/issues/1246#issuecomment-583498263_

### How has this code been tested?


* Install a stack using no source:
  `kubectl crossplane stack install --cluster --namespace crossplane-system crossplane/stack-aws:v0.5.0 stack-aws`
  * Observe the deployment in the Stack resource, and observe the deployment's final controller image
* Install a stack using a source:
  `kubectl crossplane stack install --cluster --namespace crossplane-system crossplane/stack-aws:v0.5.0 stack-aws registry.hub.docker.com`
  * Observe the deployment in the Stack resource, and observe the deployment's final controller image
  
### Checklist

I have:
- [ ] Updated any relevant [documentation] and [examples].
- [ ] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplaneio/crossplane/tree/master/design/one-pager-definition-of-done.md
